### DISCUSSION TITLE
reload-dir behaviour

### DISCUSSION THREAD

**Discussion by hovi** [Kludex/uvicorn#2009] [Potential Issue]
I run my server:
`uvicorn web.asgi:application --port 8888 --reload-dir web/ --reload-dir multi/core/ --reload`

I get output (absolute path prefix stripped):
`INFO:     Will watch for changes in these directories: ['multi/core', 'doom/web']`

I change file in other directory and I get:
`WARNING:  WatchFiles detected changes in 'multi/agents/profiles/engineering/Creator.py'. Reloading...`

I explicitly stated those 2 reload-dirs so that watcher will ignore this part of my system. Is this a bug or how to correctly setup reload to only watch for those 2 directories?

**Comment by ronnix** [comment]
I discovered today that when using `watchfiles`, uvicorn will insist on secretly watching the current directory, in addition to the directories you have specified with `--reload-dir`, without telling you so in the log.

cf. https://github.com/encode/uvicorn/blob/master/uvicorn/supervisors/watchfilesreload.py#L68-L72

PS: this issue was already discussed earlier here : https://github.com/encode/uvicorn/discussions/1783 and https://github.com/encode/uvicorn/discussions/1833
